db/db_impl.h

```diff
@@ -992,9 +992,14 @@ class DBImpl : public DB {
       writer = nullptr;
       return w;
     }
-    void ClearWriter() {
+    Status ClearWriter() {
+      Status s = writer->WriteBuffer();
```

Comment:
We were already calling WriteBuffer() in the destructor but I guess the problem was that we could not react to the return status. Given that now we do it before calling the destructor, should we remove that from the destructor and perhaps add an assert to ensure that the buffer is indeed empty when destructor is called?

Reply:
Upon further inspection, this area looks somewhat messy. There are places where the code calls Sync()/Close() on the file directly rather than going through WriteBuffer(). I believe this needs to be cleaned up and have one way of flushing log files. I think we need to do that before we replace the call to WriteBuffer() with as assert.